android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt

```diff
@@ -267,9 +269,21 @@ constructor(
   fun retrieveLastSyncTimestamp(): String? =
     sharedPreferencesHelper.read(SharedPreferenceKey.LAST_SYNC_TIMESTAMP.name, null)
 
+  fun triggerSync() {
+    viewModelScope.launch { syncBroadcaster.runFirstTimeSync() }
+  }
+
   /** This function is used to schedule tasks that are intended to run periodically */
   fun schedulePeriodicJobs() {
     workManager.run {
+      schedulePeriodically<AppSyncWorker>(
+        workId = AppSyncWorker.WORK_ID,
+        repeatInterval = applicationConfiguration.syncInterval,
+        timeUnit = TimeUnit.MINUTES,
+        requiresNetwork = true,
+        initialDelay = applicationConfiguration.syncInterval,
+      )
```

Comment:
There are instances where we would not want to schedule sync until locations for syncing data are selected. This is a use case for EUSM project, which is why the sync job was not included in this method.

Reply:
Noted. Let me review this

Reply:
Also, how do we react to the sync statuses using this approach? We need to collect the flow items. That is why the functionality is in the `SyncBroadcaster` class.

Reply:
To ensure that all sync listeners are notified.